scripts/codegen-cosmos-sdk.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+set -o errexit -o nounset -o pipefail
+command -v shellcheck >/dev/null && shellcheck "$0"
+
+OUT_DIR="./src"
+COSMOS_SDK_DIR="./cosmos-sdk-0.44/proto"
+COSMOS_SDK_THIRD_PARTY_DIR="./cosmos-sdk-0.44/third_party/proto"
+
+PLUGIN_PATH="$(realpath ./bin)/protoc-gen-ts_proto_yarn_2"
+
+mkdir -p "$OUT_DIR"
+
+protoc \
+  --plugin="$PLUGIN_PATH" \
+  --ts_proto_yarn_2_out="$OUT_DIR" \
+  --proto_path="$COSMOS_SDK_DIR" \
+  --proto_path="$COSMOS_SDK_THIRD_PARTY_DIR" \
+  --ts_proto_yarn_2_opt="esModuleInterop=true,forceLong=long,useOptionals=true" \
+  "$COSMOS_SDK_DIR/cosmos/auth/v1beta1/auth.proto" \
+  "$COSMOS_SDK_DIR/cosmos/auth/v1beta1/genesis.proto" \
+  "$COSMOS_SDK_DIR/cosmos/auth/v1beta1/query.proto" \
```

Comment:
Would also be nice to add the `authz` module here:
```suggestion
"$COSMOS_SDK_DIR/cosmos/auth/v1beta1/query.proto" \
"$COSMOS_SDK_DIR/cosmos/authz/v1beta1/authz.proto" \
"$COSMOS_SDK_DIR/cosmos/authz/v1beta1/query.proto" \
"$COSMOS_SDK_DIR/cosmos/authz/v1beta1/tx.proto" \
```

Reply:
I can also do it in a separate PR